ipsets/ipsets.go

```diff
@@ -792,6 +820,27 @@ func (s *IPSets) ApplyDeletions() {
 	s.gaugeNumIpsets.Set(float64(len(s.ipSetIDToIPSet)))
 }
 
+// tryTempIPSetDeletions tries to delete any temporary IP sets found by the last resync.
+func (s *IPSets) tryTempIPSetDeletions() {
+	s.pendingTempIPSetDeletions.Iter(func(item interface{}) error {
+		setName := item.(string)
+		logCxt := s.logCxt.WithField("setName", setName)
+		if s.existingIPSetNames.Contains(setName) {
+			logCxt.Info("Deleting IP set.")
+			if err := s.deleteIPSet(setName); err != nil {
+				// Log and carry on; we'll try again in ApplyDeletions().
+				logCxt.WithError(err).Warning("Failed to delete temporary IP set. Will retry...")
+			}
+		} else {
+			// Success! Remove from the main pending deletions set too.
```

Comment:
This comment looks misplaced, or else the two possible `else` branches here are confused.  *This* `else` branch is for `// The IP set that we meant to delete has already disappeared (somehow).`, and it looks like we're completely missing the `else` branch for `// Successfully cleaned up temporary IP set.`

Reply:
Yes, you're right, added the else to the wrong conditional, doh!